File: api/src/main/java/dev/ulloasp/mlsuite/plugin/application/dto/PluginDto.java
package dev.ulloasp.mlsuite.plugin.application.dto;

import java.time.OffsetDateTime;

public record PluginDto(
        String id,
        String fileName,
        String contentType,
        long sizeBytes,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt,
        String updatedByName,
        String updatedByEmail,
        String updatedByAvatarUrl,
        String source,
        String pluginType,
        String kind) {
}

File: frontend/src/api/plugins/dtos/plugin-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

export interface PluginDto {
  id: string;
  fileName: string;
  contentType: string;
  sizeBytes: number;
  createdAt: string;
  updatedAt: string;
  source: string;
  pluginType: "field" | "report" | "invalid";
  kind: string | null;
}
